CPAC/connectome/connectivity_matrix.py

```diff
@@ -0,0 +1,274 @@
+    """  # pylint: disable=line-too-long  # noqa E501
+    desc_regex = re.compile(r'\_desc.*(?=\_)')
+
+    data = np.genfromtxt(time_series).T
+
+    existing_desc = desc_regex.search(time_series)
+    if hasattr(existing_desc, 'group'):
+        matrix_file = time_series.replace(
+            existing_desc.group(),
+            '+'.join([
+                existing_desc.group(),
+                method
+            ])
+        ).replace('_timeseries.1D', '_connectome.npy')
+    else:
+        matrix_file = _connectome_name(time_series, method)
+
+    method = _get_method(method, 'Nilearn')
+    if method is NotImplemented:
+        return NotImplemented
+
+    connectivity_measure = ConnectivityMeasure(kind=method)
+    connectome = connectivity_measure.fit_transform([data])[0]
+
+    np.save(matrix_file, connectome)
+    return matrix_file
+
+
+def compute_connectome_nilearn(parcellation, timeseries, method):
+    """Function to compute a connectome matrix using Nilearn
+
+    Parameters
+    ----------
+    parcelation : Niimg-like object
+        http://nilearn.github.io/manipulating_images/input_output.html#niimg-like-objects
+        Region definitions, as one image of labels.
+
+    timeseries : str
+        path to timeseries image
+
+    method: str
+        'Pearson' or 'Partial'
+
+    Returns
+    -------
+    numpy.ndarray or NotImplemented
+    """
+    output = _connectome_name(timeseries, f'Nilearn{method}')
+    method = _get_method(method, 'Nilearn')
+    if method is NotImplemented:
+        return NotImplemented
+    masker = NiftiLabelsMasker(labels_img=parcellation,
+                               standardize=True,
+                               verbose=True)
+    timeser = masker.fit_transform(timeseries)
+    correlation_measure = ConnectivityMeasure(kind=method)
+    corr_matrix = correlation_measure.fit_transform([timeser])[0]
+    np.fill_diagonal(corr_matrix, 0)
+    np.savetxt(output, corr_matrix)
+    return corr_matrix
+
+
+def create_connectome(name='connectome'):
+
+    wf = pe.Workflow(name=name)
+
+    inputspec = pe.Node(
+        util.IdentityInterface(fields=[
+            'timeseries',
+            'measure'
+        ]),
+        name='inputspec'
+    )
+
+    outputspec = pe.Node(
+        util.IdentityInterface(fields=[
+            'connectome',
+        ]),
+        name='outputspec'
+    )
+
+    node = pe.Node(Function(input_names=['timeseries', 'measure'],
+                            output_names=['connectome'],
+                            function=compute_correlation,
+                            as_module=True),
+                   name='connectome')
+
+    wf.connect([
+        (inputspec, node, [('timeseries', 'timeseries')]),
+        (inputspec, node, [('measure', 'measure')]),
+        (node, outputspec, [('connectome', 'connectome')]),
+    ])
+
+    return wf
```

Comment:
This was already in C-PAC: https://github.com/FCP-INDI/C-PAC/blob/fc5971a0a4d9a800dd353a81de58d89de673ac3f/CPAC/connectome/pipeline.py#L33-L64